File: ruby/problem2.rb
fib = Fiber.new do
  x, y = 1, 1
  while true
    Fiber.yield x
    x, y = x + y, x
  end
end

count = 0
idx = 1
while true
  ans = fib.resume
  break if ans >= 4000000
  count += ans if ans%2==0
end

p count

File: golang/problem2.go
package main

import "fmt"


func fib(x, y int) int {
	if x >= 4000000 {
		return 0
	}
	if x % 2 == 0 {
		return x + fib(x+y, x)
	}
	return fib(x+y, x)
}


func main() {
	fmt.Println(fib(1, 1))
}